stm/vio_checking.ml

```diff
@@ -28,15 +28,12 @@ module Pool = Map.Make(IntOT)
 let schedule_vio_checking j fs =
   if j < 1 then CErrors.user_err Pp.(str "The number of workers must be bigger than 0");
   let jobs = ref [] in
-  List.iter (fun f ->
-    let f =
-      if Filename.check_suffix f ".vio" then Filename.chop_extension f
-      else f in
-    let long_f_dot_v, _,_,_,_, tasks, _ = Library.load_library_todo f in
-    Stm.set_compilation_hints long_f_dot_v;
+  List.iter (fun long_f_dot_vio ->
+    let _,_,_,_, tasks, _ = Library.load_library_todo long_f_dot_vio in
+    Stm.set_compilation_hints long_f_dot_vio;
```

Comment:
Shouldn't one pass to this function the .v file (rather than the the .vio)? At least this is my understanding of the old code (that was ugly, I much prefer this one, but the semantics seems to have changed).

Reply:
The PR also changes `Library.load_library_todo` so that it does not modify the path name given to it. As for `Stm.set_compilation_hints`, it uses `Aux_file.load_aux_file_for` which is indifferent about the exact suffix. Sp, it should be ok afaicj.